nexus/db-model/src/instance.rs

```diff
@@ -179,6 +179,18 @@ pub struct InstanceRuntimeState {
     /// This field is guarded by the instance's `gen`.
     #[diesel(column_name = migration_id)]
     pub migration_id: Option<Uuid>,
+
+    /// The UUID of the saga currently holding the update lock on this instance.
+    /// If this is [`None`] the instance is not locked. Otherwise, if this is
+    /// [`Some`], this contains the UUID of the saga holding the lock.
+    ///
+    /// This field is guarded by the instance's `updater_gen`
+    #[diesel(column_name = updater_id)]
+    pub updater_id: Option<Uuid>,
+
+    /// The generation number for the updater lock.
+    #[diesel(column_name = updater_gen)]
+    pub updater_gen: Generation,
```

Comment:
Can we add docs specifying that this is incremented when the lock is *successfully* acquired or released?

(On first read, I thought this was "on acquire only", but that's not the case!)